tests/test_tracing.py

```diff
@@ -0,0 +1,523 @@
+                            body=mock_server.hostport,
+                        )
+                    elif encoding == 'thrift':
+                        if endpoint == 'thrift1':
+                            response_future = tchannel.thrift(
+                                thrift_service.X.thrift1(mock_server.hostport),
+                                hostport=mock_server.hostport,
+                            )
+                        elif endpoint == 'thrift3':
+                            response_future = tchannel.thrift(
+                                thrift_service.X.thrift3(mock_server.hostport),
+                                hostport=mock_server.hostport,
+                            )
+                        elif endpoint == 'thrift4':
+                            response_future = tchannel.thrift(
+                                thrift_service.X.thrift4(mock_server.hostport),
+                                hostport=mock_server.hostport,
+                            )
+                        else:
+                            raise ValueError('wrong endpoint %s' % endpoint)
+                    else:
+                        raise ValueError('wrong encoding %s' % encoding)
+                elif transport == 'http':
+                    response_future = http_client.fetch(
+                        request=HTTPRequest(
+                            url='%s%s' % (base_url, endpoint),
+                            method='POST',
+                            body=mock_server.hostport,
+                        )
+                    )
+                else:
+                    raise NotImplementedError(
+                        'unknown transport %s' % transport)
+            response = yield response_future
+
+    body = response.body
+    if expect_baggage:
+        assert body == baggage
+
+    def get_sampled_spans():
+        return [s for s in tracer.reporter.get_spans() if s.is_sampled]
+
+    # Sometimes the test runs into weird race condition where the
+    # after_send_response() hook is executed, but the span is not yet
+    # recorded. To prevent flaky test runs we check and wait until
+    # all spans are recorded, for up to 1 second.
+    for i in range(0, 1000):
+        spans = get_sampled_spans()
+        if len(spans) >= expect_spans:
+            break
+        yield tornado.gen.sleep(0.001)  # yield execution and sleep for 1ms
+
+    spans = get_sampled_spans()
+    assert expect_spans == len(spans), 'Unexpected number of spans reported'
+    # We expect all trace IDs in collected spans to be the same
+    if expect_spans > 0:
+        spans = tracer.reporter.get_spans()
+        assert 1 == len(set([s.trace_id for s in spans])), \
+            'all spans must have the same trace_id'
+
+
+@pytest.mark.parametrize('encoding,operation', [
+    ('json', 'foo'),
+    ('thrift', 'X::thrift2'),
+])
+@pytest.mark.gen_test
+def __test_span_tags(encoding, operation, tracer, thrift_service):
```

Comment:
oh, good catch, I did this to disable them temporarily when troubleshooting the failure in the other one.